File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:

        l = 0

        maxx = 0

        sett = set()

        for r in range(len(s)):

            while s[r] in sett:

                sett.remove(s[l])

                l += 1

            sett.add(s[r])

            maxx = max(len(sett),maxx)

            

        return maxx
        

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int maxx = 0;
        unordered_set<char> sett;
        int left = 0;
        for (int right=0;right<s.size();right++){
            while (sett.find(s[right])!=sett.end()){
                sett.erase(s[left]);
                left++;
            }
            sett.insert(s[right]);
            maxx = max(maxx,int(sett.size()));
        }
        return maxx;
    }
};